sdk/storage/common/storage_checksum.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+namespace azure { namespace storage { namespace common {
+
+enum checksum_type
+{
+    none,
+    md5,
+    sha256,
+    crc64
+};
+
+class checksum
```

Comment:
Mark: We need to discuss the internal interface for checksum calculation. THE current SDK's interface works but is a bit difficult to understand.

Reply:
I think that's core sdk's business.